packages/core/src/indexing/local-index-api.ts

```diff
@@ -174,19 +173,12 @@ export class LocalIndexApi implements IndexApi {
   }
 
   async init(): Promise<void> {
-    await this.databaseIndexApi?.init()
-    // FIXME: CDB-2132 - Fragile DatabaseApi initialisation
-    await this.populateDatabaseApiInternalState()
-  }
-
-  /*
-  Gets currently indexed models from the database api and calls index's api indexModels()
-  which creates index model args and passes them back to the database api, so that it can populate
-  its internal state
-   */
-  async populateDatabaseApiInternalState(): Promise<void> {
-    const modelsToIndex = this.databaseIndexApi?.getActiveModelsToIndex()
-    await this.indexModels(modelsToIndex)
+    if (this.databaseIndexApi) {
+      await this.databaseIndexApi.init()
+      // FIXME: CDB-2132 - Fragile DatabaseApi initialisation
```

Comment:
```suggestion
      // Load the set of indexed models from the database and pass them
      // back to the DatabaseIndexApi so that it can populate its internal state.
      // TODO(CDB-2132):  Fix this fragile and circular DatabaseApi initialization
```